src/components/ButtonOption/index.js

```diff
@@ -0,0 +1,59 @@
+import React, { useContext } from 'react';
+import PropTypes from 'prop-types';
+import { ButtonGroupPickerContext } from '../ButtonGroupPicker/context';
+import HiddenElement from '../Structural/hiddenElement';
+import StyledLabel from './styled/label';
+import StyledText from './styled/text';
+import isOptionSelected from './helpers/isOptionSelected';
+import { uniqueId } from '../../libs/utils';
+
+export default function ButtonOption(props) {
+    const inputId = uniqueId('button-option');
+    const { className, style, name, label, disabled, onClick } = props;
+
+    const context = useContext(ButtonGroupPickerContext);
```

Comment:
here you can:
```const { value, type, name ... } = useContext(ButtonGroupPickerContext);```
in order to avoid repeating context. each time a prop from the context is used